packages/angular/cli/commands/update-impl.ts

```diff
@@ -550,6 +550,28 @@ export class UpdateCommand extends Command<UpdateCommandSchema> {
         return 1;
       }
 
+      if (pkg.name === '@angular/cli') {
```

Comment:
I think this check should be moved down to the next section that retrieves the manifest for the version that will be installed. `manifest.version` (the exact version that will be installed) and `node.package.version` (the exact version currently installed) can then be used within the logic.
Right above the following would probably be a good spot: https://github.com/angular/angular-cli/blob/f77aef238a4f65f4613d87562673f7d2f4343fe0/packages/angular/cli/commands/update-impl.ts#L648